.github/workflows/test.yml

```diff
@@ -18,22 +18,29 @@ jobs:
     - name: 'Test syntax check'
       run: 'rbs parse gems/**/*.rbs'
   test_on_changed_gems:
+    if: github.event_name == 'pull_request'
     runs-on: ubuntu-latest
-    container:
-      image: rubylang/ruby:3.1-focal
     steps:
     - uses: actions/checkout@v3
+    - uses: ruby/setup-ruby@v1
       with:
-        fetch-depth: 100
-    - name: 'Install dependencies'
-      run: 'bundle install --frozen'
-    - name: Get changed files
-      id: changed-files
-      uses: tj-actions/changed-files@v34.4.0
+        ruby-version: '3.1'
+        bundler-cache: true
+    - name: Get changed gems
+      id: changes
+      run: |
+        echo 'gems<<GEMS' >> $GITHUB_OUTPUT
+        gh pr diff ${{ github.event.number }} --name-only | \
+          grep '^gems/' | \
+          awk -F / '{printf "%s/%s/%s\n", $1, $2, $3}' | \
+          sort -u | \
+          tee -a $GITHUB_OUTPUT
+        echo 'GEMS' >> $GITHUB_OUTPUT
+      env:
+        GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
     - name: 'Run test on changed gems'
+      if: steps.changes.outputs.gems
```

Comment:
I extracted `if` conditional as [expressions](https://docs.github.com/en/actions/learn-github-actions/expressions). According to the document, an empty string will be converted to `0`, which means `false`.